libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -705,6 +762,135 @@ void AP_UAVCAN::do_cyclic(void)
         led_out_sem_give();
     }
 
+    _tunnel_sem->take_blocking();
+    tunnel_send();
+    _tunnel_sem->give();
+}
+
+bool AP_UAVCAN::tunnel_sem_take()
+{
+    bool sem_ret = _tunnel_sem->take(10);
```

Comment:
please use the new take_blocking() method

Reply:
and then maybe remove the tunnel_sem_take() and give methods

Reply:
Already done. I left these accessors in by accident, they are old. Look a couple lines up, I'm not calling them.
I'll remove